elyra/pipeline/registry.py

```diff
@@ -0,0 +1,800 @@
+
+        return component_parameters
+
+    def build_parameter(self, obj, obj_type):
+        data_object = {}
+        # Determine whether parameter is optional
+        if ("optional" in obj and not obj['optional']) \
+                or ("description" in obj and "required" in obj['description'].lower()):
+            data_object['required'] = True
+        else:
+            data_object['required'] = False
+
+        # Set description
+        parameter_description = ""
+        if "description" in obj:
+            parameter_description = obj['description']
+
+        # Assign type, default to string
+        data_object['format'] = "string"
+        custom_control_id = "StringControl"
+        if "type" in obj:
+            data_object['format'] = obj['type']
+            custom_control_id = self.get_custom_control_id(obj['type'].lower())
+
+        # Build label name
+        label = f"{obj['name']} ({obj_type})"
+
+        # Build parameter info
+        new_parameter = self.compose_parameter(obj['name'], custom_control_id, label,
+                                               parameter_description, data_object)
+
+        return new_parameter
+
+    def parse_component_execution_instructions(self, component_body):
+        pass
+
+
+class AirflowComponentParser(ComponentParser):
+    _type = "airflow"
+
+    def __init__(self):
+        super().__init__()
+
+    def parse_component_details(self, component_body, component_name=None):
+        # Component_body never used, but component_name never used in KFP parser
+
+        label = component_name
+
+        # TODO: Is there any way to reliably get the description for the operator overall?
+        # Could maybe pull from a class description but this wouldn't work for operators
+        # with multiple classes or those without docstrings at all.
+        description = ""
+
+        component_json = {
+            'label': label,
+            'image': "",
+            'id': get_id_from_name(label),
+            'description': ' '.join(description.split()),
+            'node_types': []
+        }
+
+        node_type = set_node_type_data(get_id_from_name(label), label, description)
+        component_json['node_types'].append(node_type)
+
+        return component_json
+
+    def parse_component_properties(self, component_body, component_path):
+        '''
+        Build the properties object according to the operator python file and return properties.
+        '''
+        # Start with empty properties object
+        component_parameters = copy.deepcopy(empty_properties)
+
+        # Add path details
+        component_parameters['current_parameters']['component_source'] = component_path
+
+        # Add class information as parameter
+        component_parameters['parameters'].append({"id": "elyra_airflow_class_name"})
+        component_parameters['current_parameters']["elyra_airflow_class_name"] = []
```

Comment:
What is the reason for this defaulting? to an empty array?  It seems contrary to the 'For Airflow inputs, remove class name information from key' logic in `PipelineParser._get_remaining_component_params` which expects a string.

Where does the actual value of `elyra_airflow_class_name` come from?

Reply:
This is part of the way I'm handling an operator file that includes more than one operator class. [This](https://github.com/apache/airflow/blob/1.10.15/airflow/operators/slack_operator.py) is a good example of a file that has 2 possible operators to use. 

So the `elyra_airflow_class_name` var holds an array of classes from which the user can choose in the properties pane (as in the screenshot below). The user will select this class (I suppose I should make it `required` now that I think about it), and the selected class will be propagated as a parameter to the backend where the parser will then use to determine which of the remaining parameters it needs to include as kwargs for the selected class.

<img width="459" alt="Screen Shot 2021-06-07 at 3 06 51 PM" src="https://user-images.githubusercontent.com/31816267/121081205-fe987b00-c7a1-11eb-915a-7b881d685bfd.png">

That being said, I'm not sure I love this approach for the long term, but this was a fairly simple way to include that functionality.